lua/entities/gmod_wire_expression2/base/tokenizer.lua

```diff
@@ -280,27 +282,62 @@ function Tokenizer:Next()
 				if m:sub( -1, -1) == "\"" then
 					break
 				else -- Escape
-					local c = self:At()
-
-					if not Escapes[c] then
-						self:Warning("Invalid escape \\" .. c)
-						c = '\\' .. c
+					local char = self:At()
+					local esc, err = ""
+
+					-- Using %g here just to be a bit more informative on warnings
+					if escapes[char] then
+						self:NextChar()
+						esc = escapes[char]
+					elseif char == "u" then
+						self:NextChar()
+
+						if self:At() ~= "{" then err = "Unicode escape must begin with {" goto _err end
+
+						esc = self:ConsumePattern("^%b{}", true)
+
+						if not esc then err = "Unicode escape must end with }"
+						elseif #esc == 2 then err = "Unicode escape cannot be empty"
+						elseif #esc > 8 then err = "Unicode escape can only contain up to 6 characters"
+						else
+							local num = tonumber(esc:sub(2, -2), 16)
+							if not num then
+								err = "Unicode escape must contain hexadecimal digits"
+							elseif num >= 0x10ffff then
+								err = "Unicode escape cannot be greater than 10ffff"
+							else
+								esc = utf8.char(num)
```

Comment:
Maybe worth keeping this as a utility local function, the garry implementation is suboptimal since it has support for passing multiple numbers and I just want to avoid gmod functions

https://github.com/Facepunch/garrysmod/blob/e189f14c088298ca800136fcfcfaf5d8535b6648/garrysmod/lua/includes/modules/utf8.lua#L104-L138

I suppose this would work (just inlined everything and removed multiple number support)
```lua
local function toUnicodeChar(v)
		if v < 0x80 then -- Single-byte sequence
			return string.char(v)
		elseif v < 0x800 then -- Two-byte sequence
            return string.char(
			    bit.bor( 0x80, bit.band( bit.rshift( v, 6 ), 0x3F ) ),
			    bit.bor( 0x80, bit.band( v, 0x3F ) )
			 )
		elseif v < 0x10000 then -- Three-byte sequence
             return string.char(
			    bit.bor( 0x80, bit.band( bit.rshift( v, 12 ), 0x3F ) ),
			    bit.bor( 0x80, bit.band( bit.rshift( v, 6 ), 0x3F ) ),
			    bit.bor( 0x80, bit.band( v, 0x3F ) )
			 )
		else -- Four-byte sequence
			return string.char(
			    bit.bor( 0xF0, bit.band( bit.rshift( v, 18 ), 0x07 ) ),
			    bit.bor( 0x80, bit.band( bit.rshift( v, 12 ), 0x3F ) ),
			    bit.bor( 0x80, bit.band( bit.rshift( v, 6 ), 0x3F ) ),
			    bit.bor( 0x80, bit.band( v, 0x3F ) )
			 )
		end
end

Reply:
This would be a lot sweeter as a Wirelib function.